### PR TITLE
Add worker_class argument to LocalCluster

### PR DESCRIPTION
**Pull request by pentschev**
Allows instantiating workers subclassed from Worker (e.g., CUDAWorker) in LocalCluster.

@mrocklin

### CODE REVIEW

**Comment by pentschev** [comment]
It looks like the failures are not related to this PR, but I'm not quite sure about it.

File: distributed/deploy/local.py [hunk 1]
```diff
@@ -203,6 +206,8 @@ def __init__(
         if security:
             self.worker_kwargs["security"] = security
 
+        self.worker_class = worker_class
```
Comment (mrocklin) [inline distributed/deploy/local.py]:
```suggestion
        if not worker_class:
            if processes:
                worker_class = Nanny
            else:
                worker_class = Worker
        self.worker_class = worker_class
```

File: distributed/deploy/local.py [hunk 2]
```diff
@@ -280,9 +285,10 @@ def _start_worker(self, death_timeout=60, **kwargs):
 
         if self.processes:
             W = Nanny
+            kwargs["worker_class"] = self.worker_class
             kwargs["quiet"] = True
         else:
-            W = Worker
+            W = self.worker_class or Worker
 
         w = yield W(
```
Comment (mrocklin) [inline distributed/deploy/local.py]:
```suggestion
        w = yield self.worker_class(
```

File: distributed/deploy/local.py [hunk 3]
```diff
@@ -280,9 +285,10 @@ def _start_worker(self, death_timeout=60, **kwargs):
 
         if self.processes:
             W = Nanny
+            kwargs["worker_class"] = self.worker_class
             kwargs["quiet"] = True
         else:
-            W = Worker
+            W = self.worker_class or Worker
```
Comment (mrocklin) [inline distributed/deploy/local.py]: If you accept the other changes, then we can probably drop the conditional above (it's effectively now handled in the constructor)

We might also remove the `self.processes` attribute, which seems to be effectively replaced and generalized by `self.worker_class`.

We should have something like the following though to maintain previous behavior.

```python
if self.worker_class is Nanny:
    kwargs['quiet'] = True
```

Comment (pentschev) [reply distributed/deploy/local.py]: Sure, your changes look fine to me. I will push a commit.

**Comment by mrocklin** [comment]
This could also use a test.  Maybe something like ...

```python
def test_worker_class(loop):
    class MyWorker(Worker):
        pass

    with LocalCluster(n_workers=2, loop=loop, worker_class=MyWorker, scheduler_port=0, dashboard_address=None) as cluster:
        assert all(isinstance(w, MyWorker) for w in cluster.workers)
```

**Comment by pentschev** [comment]
Thanks for the suggestion, I was really unsure how to test this. I extended your test a bit to test both `Worker` and `Nanny`.

**Comment by mrocklin** [comment]
I should have mentioned that we recently switched to using `black` for code styling.  It looks like you found that out.  I need to add docs for this in a more prominent place, but you may want to try installing [pre commit hooks](https://ml.dask.org/contributing.html#pre-commit-hooks)

**Comment by pentschev** [comment]
> I should have mentioned that we recently switched to using `black` for code styling. It looks like you found that out. I need to add docs for this in a more prominent place, but you may want to try installing [pre commit hooks](https://ml.dask.org/contributing.html#pre-commit-hooks)

Ah nice, didn't know about this, thanks for that!

I think the errors are not related to this PR.

**Comment by pentschev** [comment]
No worries, let me know if you need help with them.

**Pull request merged**